pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -564,14 +568,27 @@ func buildOutboundTCPFilter(mesh *meshconfig.MeshConfig, attrsIn attributes, nod
 	}
 
 	if util.IsXDSMarshalingToAnyEnabled(node) {
-		out.ConfigType = &listener.Filter_TypedConfig{TypedConfig: util.MessageToAny(cfg)}
+		out.ConfigType = &listener.Filter_TypedConfig{TypedConfig: gogo.AnyToProtoAny(util.MessageToGogoAny(cfg))}
 	} else {
 		out.ConfigType = &listener.Filter_Config{Config: util.MessageToStruct(cfg)}
 	}
 
 	return out
 }
 
+// XDS protos use golang/protobuf, but the istio APIs use gogo protos. This means the golang/protobuf
+// backend needs to register the mixer client filter protos.
+func init() {
+	//nolint: lll
+	proto.RegisterEnum("istio.mixer.v1.config.client.NetworkFailPolicy_FailPolicy", mccpb.NetworkFailPolicy_FailPolicy_name, mccpb.NetworkFailPolicy_FailPolicy_value)
```

Comment:
Is this needed for serializing to Any?

Reply:
Otherwise we get "unknown message" errors. not sure if there is a better alternative... I don't like this either.